activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -63,8 +63,8 @@ def locking_enabled? #:nodoc:
 
         def increment_lock
           lock_col = self.class.locking_column
-          previous_lock_value = send(lock_col).to_i
-          send(lock_col + "=", previous_lock_value + 1)
+          previous_lock_value = _read_attribute(lock_col)
```

Comment:
Is not the `to_i` to make `nil` to `0`? Maybe we don't have test coverage for this case but it seems to be the case.